packages/clerk-js/src/core/resources/User.ts

```diff
@@ -241,7 +241,10 @@ export class User extends BaseResource implements UserResource {
   };
 
   delete = (): Promise<void> => {
-    return this._baseDelete({ path: '/me' });
+    return this._baseDelete({ path: '/me' }).then(res => {
+      User.clerk.__internal_broadcastSignOutEvent();
```

Comment:
this would be much better with the event emitter in use, no unnecessary awareness of `Clerk` here.